smoke-test/src/main/java/org/opennms/smoketest/containers/MinionContainer.java

```diff
@@ -92,7 +94,11 @@ public class MinionContainer extends GenericContainer implements KarafContainer,
     private final String location;
     private final GenericContainer container;
 
-    private MinionContainer(final StackModel model, final String id, final String location) {
+    public interface WaitStrategyInterface {
```

Comment:
Why have this interface at all? Is there any need in getting the container which will have the `container` field uninitialized? Just use `WaitStrategy` instead?

Reply:
If you opt for keeping it, it should be named `WaitStrategyProvider` or something.

Reply:
Or maybe just a `Function<MinionContainer, WatiStrategy>`?

Reply:
The container itself is passed to the strategy.

Reply:
done.